### PR TITLE
Parallelize 5: Add new uniform tensor implementation

### PR DESCRIPTION
**Pull request by DarshalShetty**

### CODE REVIEW

**Comment by themetaschemer** [comment]
Needs rebase, I think

**Comment by DarshalShetty** [comment]
Rebase done

File: uniform-tensors/tensors/D-extend.rkt
Comment (themetaschemer) [inline uniform-tensors/tensors/D-extend.rkt:193]: Get rid of commented lines.

File: uniform-tensors/tensors/D-extend.rkt
Comment (themetaschemer) [inline uniform-tensors/tensors/D-extend.rkt:194]: No nested defines in our coding conventions :) Use a let. 

Same two comments on line 216 and 217

File: uniform-tensors/tensors/0-vectors.rkt
Comment (themetaschemer) [inline uniform-tensors/tensors/0-vectors.rkt:51]: Extra blank line

**Review by themetaschemer — APPROVED** [review]
Approved over all, but please comments before merging.

File: uniform-tensors/tensors/0-vectors.rkt
Comment (DarshalShetty) [reply uniform-tensors/tensors/0-vectors.rkt:51]: Should I remove the blank line before the `print-vec` definition?

Comment (themetaschemer) [reply uniform-tensors/tensors/0-vectors.rkt:51]: Sorry, I must have misread it. No, this is fine.

**Comment by DarshalShetty** [comment]
I was thinking of doing a "rebase and merge" for this PR instead of "squash and merge". Do you agree @themetaschemer ?

**Comment by themetaschemer** [comment]
Yes. Agreed. But clean up the commit history using an interactive rebase so that superflous commits (like correcting indentation and typos) don't become part of the commit history. 
    On Saturday, November 2, 2024 at 10:58:32 AM PDT, Darshal Shetty ***@***.***> wrote:  
 
 


I was thinking of doing a "rebase and merge" for this PR instead of "squash and merge". Do you agree @themetaschemer ?

**Comment by DarshalShetty** [comment]
Oh, sorry about the "Add indentation" commit that has already been pushed to main. I could reset that commit and squash it with one of the commits in this branch. That would cause me to force push to the main branch though.

I have reworded 2 commits on this branch and force pushed them. I'll merge without squashing once you approve it.

**Pull request merged**
